internal/agent/list.go

```diff
@@ -0,0 +1,94 @@
+package agent
+
+import (
+	"fmt"
+
+	"github.com/buildkite/cli/v3/pkg/cmd/factory"
+	"github.com/buildkite/go-buildkite/v3/buildkite"
+	"github.com/charmbracelet/bubbles/list"
+	tea "github.com/charmbracelet/bubbletea"
+	"github.com/charmbracelet/lipgloss"
+)
+
+var agentListStyle = lipgloss.NewStyle().Margin(1, 2)
+
+type agentListItem struct {
+	title, metadata string
+}
+
+func (ali agentListItem) Title() string       { return ali.title }
+func (ali agentListItem) Description() string { return ali.metadata }
+func (ali agentListItem) FilterValue() string { return ali.title }
+
+type agentListModel struct {
+	agentList list.Model
+	agents    []buildkite.Agent
+	quitting  bool
+}
+
+func ObtainAgents(f *factory.Factory, args ...string) (*agentListModel, error) {
```

Comment:
I am not sure if it is safe to use variadic functions when we don't expect the arguments to be variable.

Reply:
Was on the fence with it - this varidic function implantation should allow for 0:N arguments to be passed in (the 3 flag values are defaulted to empty strings to cater for the use!)

Reply:
Another thing, the args are not interchangeable in the implementation but the use of variadic func make it seem like they are.

Reply:
Yeah I think named args are a safer approach. we can always add to later if we need more

Reply:
Switched back 👍